attest/eventlog.go

```diff
@@ -564,20 +570,25 @@ func parseRawEvent2(r *bytes.Buffer, specID *specIDEvent) (event rawEvent, err e
 		if err := binary.Read(r, binary.LittleEndian, &algID); err != nil {
 			return event, err
 		}
-		var digest []byte
+		var digest digest
+
 		for _, alg := range specID.algs {
 			if alg.ID != algID {
 				continue
 			}
 			if uint16(r.Len()) < alg.Size {
 				return event, fmt.Errorf("reading digest: %v", io.ErrUnexpectedEOF)
 			}
-			digest = make([]byte, alg.Size)
+			digest.data = make([]byte, alg.Size)
+			hash, ok := algorithmMap[tpm2.Algorithm(alg.ID)]
```

Comment:
I think you can just do `HashAlg(tpm2.Algorithm(alg.ID)).cryptoHash()`, as the values for `attest.HashAlg` and `tpm2.Algorithm` should match